packages/babel-register/src/loader.mjs

```diff
@@ -0,0 +1,287 @@
+// Many thanks to the author of https://gist.github.com/TravisMullen/16dfabca0db4bb4c32fb4f5ef297b25a
+
+/* eslint-disable import/extensions */
+
+// this is a workarround to make sure global._BABEL_ESM_REGISTER is set to true before anything else gets loaded
+import { readFile, access } from "node:fs/promises";
```

Comment:
https://github.com/nodejs/node/issues/38006
I personally prefer to use the sync version, because it is more convenient to use and there is no obvious performance loss.
NOTE: This is just my opinion, I'm not sure what other reviewers think.

Reply:
The async version allows Node.js to call out loader with multiple files in parallel, but if the promise-based version is so much slower we could use the callback-based version.

Reply:
There should be little difference in the new version of nodejs.

Reply:
Initially I went with not using it because I thought existsSync was deperacted. turns out only exists is, not existsSync. But having it async if it doesn't make a difference speed-wise is better then.